base/compiler/abstractinterpretation.jl

```diff
@@ -1793,7 +1793,9 @@ function abstract_eval_global(M::Module, s::Symbol)
     if isdefined(M,s) && isconst(M,s)
         return Const(getfield(M,s))
     end
-    return Any
+    ty = ccall(:jl_binding_type, Ptr{Cvoid}, (Any, Any), M, s)
```

Comment:
This could return `nothing` instead of NULL to avoid `unsafe_pointer_to_objref`.

Reply:
Ok, I'll change that